recipes/duktape/all/conanfile.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+from conans import CMake, ConanFile, tools
+
+
+class DuktapeConan(ConanFile):
+    name = "duktape"
+    license = "MIT"
+    description = "Duktape is an embeddable Javascript engine, with a focus on portability and compact footprint."
+    topics = ("javascript", "engine", "embeddable", "compact")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://duktape.org"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        # Duktape has configure script with a number of options.
+        # However it requires python 2 and PyYAML package
+        # which is quite an unusual combination to have.
+        # The most crucial option is --dll which just flips this define.
+        if self.settings.os == "Windows" and self.options.shared:
+            tools.replace_in_file(
+                os.path.join(self._source_subfolder, "src", "duk_config.h"),
+                "#undef DUK_F_DLL_BUILD",
+                "#define DUK_F_DLL_BUILD",
+            )
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["duktape"]
+        if self.settings.os != "Windows" and not self.options.shared:
```

Comment:
```suggestion
        if self.settings.os == "Linux":
```

Reply:
what about other posix systems like FreeBDS, Solaris and MacOS?

Reply:
There is no separation for libm on those systems.

Reply:
Apparently only MacOS doesn't require `-lm`.
Tried OOC on FreeBSD 11.2:
```
/usr/local/bin/ld: /home/AzureUser/.conan/data/duktape/2.5.0/_/_/package/9d95320cfe08748601ee4b569da5707e402e2c39/lib/libduktape.a(duktape.c.o): in function `duk__js_execute_bytecode_inner.isra.0':
duktape.c:(.text.hot+0x23b2): undefined reference to `pow'
/usr/local/bin/ld: duktape.c:(.text.hot+0x2491): undefined reference to `pow'
/usr/local/bin/ld: duktape.c:(.text.hot+0x3ee9): undefined reference to `fmod'
```

IIRC there were no problem when duktape or test program were compiled in C++ mode and therefore it was not in the original PR.

duktape has some code to use C++ exceptions (instead of `setjump` / `longjump`, only internally). Then it should be possible to build it with C++ compiler (and thus no discussion about `-lm`), but I'm not convinced that it's a good default option.